serverless/aws/lambda/src/main/scala/sttp/tapir/serverless/aws/lambda/LambdaHandler.scala

```diff
@@ -0,0 +1,46 @@
+package sttp.tapir.serverless.aws.lambda
```

Comment:
Reading a bit into the code I'm not sure that putting this here is the right thing to do. More specifically, what would be the relationship of this class with `AwsLambdaRuntime`? 

As far as I understand, our `AwsLambdaRuntime` duplicates the functionality of `LambdaHandler`, which implements the same logic, but using amazon's java library. 

Since we have our own runtime - written without the need to use input/output streams - maybe instead it would make sense to use it instead of the `LambdaHandler` (wherever it's used). Unless there's some good reason to using the amazon-based implementation there. Either way, we should straighten these two up and document in `lambda.md` - so that it's clearly outlined what are the responsibilites of each module.

And if we *do* keep `LambdaHandler`, then I think it should go in the `runtime` package as it's a runtime component.
Lastly, there's also the JS implementation, where we can't depend on a java library. Hence if at all, the `LambdaHandler` would  need to end up in `scalajvm` sources.

Reply:
LambdaHandler is the result of removing duplications spotted among a few different modules (cdk, sam, terraform). All of them have already been based on amazon's java library. I have just introduced a more generic implementation that satisfies the needs of all of them. 

I was not aware of the real purpose of AwsLambdaRuntime. Now it seems to be a much better direction to go. The only problem I see is being tight to AwsResponse. The case is we have to support both versions of aws requests. But then the Route alias would get a bit dispersed.

The other arguable matter is if it would work on the amazon side. SAM/CDK requires us to provide a specific handler. I am not sure how all of that will behave if we use AwsLambdaRuntime (which does not conform to the AWS expectations)

Up to this point, each existing implementation has been tested using the AWS java library, and changing that approach should result in extensive testing and experimentation.

Reply:
Ah maybe SAM/CDK won't work with our handler. Ok, let's keep it in `lambda`, but in the jvm-only sources, and in the `runtime` package. As a follow-up task (separate PR), let's check if we can use our `AwsLambdaRuntime` exlusively.

Reply:
(and then we can scope the amazon-java dependency to jvm only so that the js version works)